dns_check/datadog_checks/dns_check/dns_check.py

```diff
@@ -34,14 +36,14 @@ class DNSCheck(NetworkCheck):
     SERVICE_CHECK_NAME = 'dns.can_resolve'
     DEFAULT_TIMEOUT = 5
 
-    def __init__(self, name, init_config, agentConfig, instances):
+    def __init__(self, name, init_config, agentConfig, instances={}):
         # Now that the DNS check is a Network check, we must provide a `name` for each
         # instance before calling NetworkCheck to make backwards compatible with old yaml.
         for idx, inst in enumerate(instances):
             try:
                 inst['name'] = inst['name']
             except KeyError:
-                inst['name'] = 'dns-check-%s' % idx
+                inst['name'] = "dns-check-{0}".format(idx)
```

Comment:
I find named/numbered param less error-prone. Is this a guideline?
In this case, it's ok to not used named params because we only have one but when we have many, not using named placeholders is not a good idea.

Reply:
Named is of course fine, it's just the indexed ones we're getting rid of.